enclave/core/switchlesscalls.c

```diff
@@ -37,40 +38,28 @@ bool oe_is_switchless_initialized()
 **
 **==============================================================================
 */
-oe_result_t oe_handle_init_switchless(uint64_t arg_in)
+oe_result_t oe_handle_init_switchless(
```

Comment:
It seems to me that this function should be TEE-specific.

On OP-TEE, there is more work to do to ensure that the buffer used to signal OCALL threads is backed by the same physical memory. More specifically, checks must be performed to ensure that shared memory is available on the system and that *it* is used for the worker contexts instead of them being mediated by copies.

Also, by creating different implementations of the initialization routine, you can make the OP-TEE one return an error for now.

Reply:
+1